framework/src/utils/MathUtils.C

```diff
@@ -235,4 +235,28 @@ pow(Real x, int e)
   return neg ? 1.0 / result : result;
 }
 
+// Evaluate value of polynomial using Horner's method
+Real
+poly(std::vector<Real> c, const Real x, const bool derivative)
+{
+  const Real size = c.size();
```

Comment:
I'd make `size` and integer (after all below you are doing integer math with it). I would also suggest to template this  on the number type to make it usable with AD in teh future.

Reply:
Whoa...you read my mind about templating without me having to ping you...

Reply:
Probably should do it to pow too.